runner/types.go

```diff
@@ -75,29 +75,41 @@ type Result struct {
 // function to get dsl variables from result struct
 func dslVariables() []string {
```

Comment:
Since we are using `github.com/mitchellh/mapstructure` to produce the map of values for the expression, can't we use the same library to retrieve the list of available key names?

Reply:
With the use of `github.com/mitchellh/mapstructure` library the fields in the result struct are being removed if it's empty, so we'll need to pass some random values to all the fields which requires hardcoding the values. Also the other issue is it's not creating a flagged map with key value , the map we get is `map[asn:map[as-country: as-name: as-number: as-range:]` which requires multiple typecasts and iterations.

Reply:
Implemented it in this way (and fixed few other issues) since we need an exact mapping between dsl variable list and mapstructure generated variable pairs